src/researchhub_comment/views/rh_comment_thread_view_mixin.py

```diff
@@ -0,0 +1,124 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.permissions import (
+    AllowAny,
+    IsAuthenticated,
+)
+
+from researchhub_comment.related_models.rh_comment_model import RhCommentModel
+from researchhub_comment.related_models.rh_comment_thread_model import (
+    RhCommentThreadModel,
+)
+
+
+class RhCommentThreadViewMixin:
+    @action(detail=True, methods=["POST"], permission_classes=[IsAuthenticated])
+    def create_rh_comment(self, request, pk=None):
+        try:
+            rh_thread = self._retrieve_or_create_thread_from_request_data(request)
+            _rh_comment = RhCommentModel.create_from_data(request.data, rh_thread)
+            rh_thread.refresh_from_db()  # object update from fresh db values
+            return Response(self.get_serializer(instance=rh_thread).data, status=200)
+        except Exception as error:
+            return Response(
+                f"Failed - create_rh_comment: {error}",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    @action(detail=True, methods=["POST"], permission_classes=[AllowAny])
+    def get_comment_threads(self, request, pk=None):
+        raise NotImplementedError("Needs to be implemented")
+
+    def _retrieve_or_create_thread_from_request_data(self, request):
+        request_data = request.data
+        try:
+            thread_id = request_data.get("thread_id") or None
+            if thread_id is not None:
+                return RhCommentThreadModel.objects.get(id=int(thread_id))
```

Comment:
No need to cast thread_id

Reply:
some incoming form data (like postman? casts int to string. better to be safe

Reply:
I don't think it has any real benefit?

For example:
1. Int
```
In [22]: User.objects.get(id=2)
Out[22]: <User: 2: bank bank>
```
2. String
```
In [23]: User.objects.get(id="2")
Out[23]: <User: 2: bank bank>
```
3. String to Int
```
In [25]: User.objects.get(id=int("2"))
Out[25]: <User: 2: bank bank>
```
4. String to Int with error
```
In [26]: User.objects.get(id=int("\asfd\adsf"))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[26], line 1
----> 1 User.objects.get(id=int("\asfd\adsf"))

ValueError: invalid literal for int() with base 10: '\x07sfd\x07dsf'
```
5. String with error
```
In [27]: User.objects.get(id="\asfd\adsf")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
   ...
   ...
   2019 except (TypeError, ValueError) as e:
-> 2020     raise e.__class__(
   2021         "Field '%s' expected a number but got %r." % (self.name, value),
   2022     ) from e

ValueError: Field 'id' expected a number but got '\x07sfd\x07dsf'.
```

Reply:
Accepted. I'll try removing int casting for these places. But for now, I'm going to keep the int cast in the ViewFilter. It was error-ing without it